daisi/src/cpps/logical/algorithms/disposition/centralized_participant.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 The SOLA authors
+//
+// This file is part of DAISI.
+//
+// DAISI is free software: you can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation; version 2.
+//
+// DAISI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+// the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along with DAISI. If not, see
+// <https://www.gnu.org/licenses/>.
+//
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include "centralized_participant.h"
+
+namespace daisi::cpps::logical {
+
+CentralizedParticipant::CentralizedParticipant(std::shared_ptr<sola_ns3::SOLAWrapperNs3> sola,
+                                               std::shared_ptr<OrderManagement> order_management)
+    : DispositionParticipant(sola),
+      order_management_(
+          std::move(std::dynamic_pointer_cast<SimpleOrderManagement>(order_management))){};
```

Comment:
Yes, order_management_ will always be a pointer onto a simple OrderManagement. The general class in the constructor is necessary because it is called by an AmrLogicalAgent which stores a pointer to an OrderManagement and not SimpleOM, because other subclasses are possible here.